applications/smart-distancing/libs/core.py

```diff
@@ -188,3 +197,90 @@ def non_max_suppression_fast(object_list, overlapThresh):
                                                   np.where(overlap > overlapThresh)[0])))
         updated_object_list = [j for i, j in enumerate(object_list) if i in pick]
         return updated_object_list
+
+
+    def calculate_distance_of_two_points_of_boxes(self,first_point, second_point):
+    
+        """
+        This function calculates a distance L for two input corresponding points of two detected bounding boxes.
+        it is assumed that each person is H = 170 cm tall in real scene to map the distances in the image (in pixels) to 
+        physical distance measures (in meters). 
+
+        params:
+        first_point: (x, y, h)-tuple, where x,y is the location of a point (center or each of 4 corners of a bounding box)
+        and h is the height of the bounding box. 
+        second_point: same tuple as first_point for the corresponding point of other box 
+
+        returns:
+        L:  Estimated physical distance (in centimeters) between first_point and second_point.
+
+
+        """
+
+        # estimate corresponding points distance
+        [xc1, yc1, h1] = first_point
+        [xc2, yc2, h2] = second_point
+        
+        Dx = xc2 - xc1
+        Dy = yc2 - yc1
+        
+        Lx = Dx * 170 * (1/h1 + 1/h2)/2
+        Ly = Dy * 170 * (1/h1 + 1/h2)/2
+        
+        L=math.sqrt(Lx**2+Ly**2)
+        
+        return L 
+
+
+    def calculate_box_distances(self, nn_out):
+        
+        """
+        This function calculates a distance matrix for detected bounding boxes.
+        Two methods are implemented to calculate the distances, first one estimates distance of center points of the
+        boxes and second one uses minimum distance of each of 4 points of bounding boxes.
+
+        params:
+        object_list: a list of dictionaries. each dictionary has attributes of a detected object such as
+        "id", "centroidReal" (a tuple of the centroid coordinates (cx,cy,w,h) of the box) and "bboxReal" (a tuple
+        of the (xmin,ymin,xmax,ymax) coordinate of the box)
+
+        returns:
+        distances: a NxN ndarray which i,j element is estimated distance between i-th and j-th bounding box in real scene (cm)
+
+        """
+
+        distances = []
+        for i in range(len(nn_out)):
+            distanceRow=[]
+            for j in range(len(nn_out)):
+                if i == j:
+                    L = 0
+                else:
+                    if ( int(self.dist_method) == 2 ):
+                        first_point_of_first_box = [nn_out[i]["bboxReal"][0],nn_out[i]["bboxReal"][1],nn_out[i]["bboxReal"][3]-nn_out[i]["bboxReal"][1]]
+                        second_point_of_first_box = [nn_out[i]["bboxReal"][2],nn_out[i]["bboxReal"][1],nn_out[i]["bboxReal"][3]-nn_out[i]["bboxReal"][1]]
```

Comment:
It is more readable if you define two variables for each object's height and use it instead of calculate them for each point of the object.

Reply:
We have each object's height already in `nn_out[i]["centroidReal"][3]`